zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -137,15 +161,73 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         );
 
         // Transfer value from caller to callee
+        // Use cb.query_word and BatchedIsZeroGadget as TransferWithGasFeeGadget
+        // expects words instead of expressions for tx_value and gas_fee
+        let effective_tx_value = cb.query_word();
+        let effective_gas_fee = cb.query_word();
+        let effective_tx_value_and_gas_fee_are_zero = BatchedIsZeroGadget::construct(
+            cb,
+            [effective_tx_value.expr(), effective_gas_fee.expr()],
+        );
+        cb.condition(is_tx_invalid.expr(), |cb| {
+            cb.require_equal(
+                "effective_tx_value == 0 && effective_gas_fee == 0",
+                effective_tx_value_and_gas_fee_are_zero.expr(),
+                true.expr(),
+            );
+        });
```

Comment:
I don't think you need to conditionally check if both are zeros, you can just force them to be `0` when `is_tx_invalid` is true.

```suggestion
        cb.condition(is_tx_invalid.expr(), |cb| {
             cb.require_equal(
                "effective_tx_value == 0",
                effective_tx_value.expr(),
                0.expr(),
            );
            cb.require_equal(
                "effective_gas_fee == 0",
                effective_gas_fee.expr(),
                0.expr(),
            );
        });
```